### PR TITLE
Layer stack

### LINKED ISSUE #8: Layer stack

### PR DESCRIPTION
**Pull request by RogueMacro**

### CODE REVIEW

File: Engine/src/LayerStack.bf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+					{
+						_layers.RemoveAt(i);
+						break;
+					}
+				}	
+			}
+			else
+			{
+				if (_layers.Count > _layerInsert)
+					_layers.RemoveAt(_layers.Count - 1);
```
Comment (Fusioon) [inline Engine/src/LayerStack.bf]: You could use `_layers.PopBack()`.

**Review by Fusioon — COMMENTED** [review]
Seems ok to me. Can we add some method to LayerStack which would remove layer by reference. Also after consideration we probably shouldn't delete Layers inside LayerStack I have already seen it cause problems yesterday. We could have it optional with some flag when adding the Layer.

**Comment by Fusioon** [comment]
I don't see the need for Layers in the Engine. this may be used for Editor but i really can't imagine how we would use this inside the engine. Maybe you could explain it to me, please?

**Comment by RogueMacro** [comment]
Actually, you've got a point there. I will move it to Editor

**Comment by RogueMacro** [comment]
After a bit of thought it might be used like this:
LayerStack
  - World (3D Scene)
  - UI (Health bar, text, etc)
  - Debug (If we wanted this. Like in unreal, a small in-game console that shows message)
  - Editor (ImGui)

Each layer would have some renderables, projection matrix/camera, and maybe a shader.

**Comment by RogueMacro** [comment]
One idea is that for people only using SteelEngine and not the editor, it would be nicer to debug stuff with imgui right? Edit values on runtime and stuff. Therefore, each layer could have the possibility of rendering its own imgui window. @Fusioon

**Review by jamesaorson — DISMISSED** [review]

**Pull request merged**